Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1888,6 +1890,28 @@ public boolean isRingBearer() {
         return ringBearerFlag;
     }
 
+    @Override
+    public boolean isSolved() {
+        return solved;
+    }
+
+    private static final String solvedInfoKey = "IS_SOLVED";
+
+    @Override
+    public void setSolved(boolean value, Game game, Ability source) {
+        if (!value || !game.replaceEvent(GameEvent.getEvent(
+                EventType.BECOME_SOLVED, getId(),
```

Comment:
If you implementing a replace event support then must use a standard logic with two types of event: 
  * become_solve (can be replaced);
  * become_solved (info event for triggers/watchers only).